메인/메인메뉴 페이지 추가

cjyeon opened the pull request.

cjyeon committed:
- 메인메뉴 및 체크팝업 jsp/js 추가
- 로그인 후 페이지 이동 수정

cjyeon committed:
- 마이페이지 jsp/js 추가

cjyeon committed:
- 메인메뉴 설정(회원탈퇴) 페이지 추가

cjyeon committed:
- 회원탈퇴/로그아웃 기능 추가

yuncorn commented on src/main/webapp/WEB-INF/jsp/members/mypage/mypage.jsp:
```diff
@@ -0,0 +1,73 @@
+<html>
+<head>
+    <jsp:include page="/WEB-INF/jsp/common/include/header.jsp">
+        <jsp:param name="title" value="MainNavigationMenu"/>
```
영역별 타이틀 다시 확인부탁드려요

wjdwo951219 commented on src/main/webapp/resources/js/members/changePassword.js:
```diff
@@ -40,11 +40,13 @@ $(function () {
     }
 
     $("#change_password_complete").on('click', function () {
+        console.log(inputPresentMemberPassword.val() + " " + inputNewMemberPassword.val());
```
제거 요청 드립니다.

wjdwo951219 commented on src/main/webapp/resources/js/members/setting/withdraw.js:
```diff
@@ -0,0 +1,61 @@
+                    if (result.withDrawSuccess === true) {
+                        popUpOpen("회원 탈퇴가 완료되었습니다.");
+                        $(".popup_close_btn").attr("onclick", null);
+                        $(".popup_close_btn").on('click', function () {
```
line 45와 method chaining으로 작성하면 jquery selector의 중복이 제거 될 것 같습니다.

wjdwo951219 replied:
method chaining 안되면 const로 선언해서 중복 제거 요청드립니다.

wjdwo951219 commented on src/main/webapp/resources/js/members/setting/withdraw.js:
```diff
@@ -0,0 +1,61 @@
+            },
+            error: function (error) {
+                isAjaxProcessing = false;
+                console.log(error);
```
제거 요청 드립니다.

yuncorn commented on src/main/java/com/tripmate/controller/MemberController.java:
```diff
@@ -316,4 +316,55 @@ private String isSendMailSuccess(final String type, final MemberMailDTO memberMa
+
+            SignInDTO withdrawDTO = SignInDTO.builder()
+                    .memberNo(sessionDTO.getMemberNo())
+                    .memberId(signInDTO.getMemberId())
```
jsp 내에서 세션의 memberInfo.memberId 값을 사용해서 컨트롤러로 그대로 넘기는 거로 보이는데
화면단에서 비밀번호만 입력 받는다면 338번라인 memberNo처럼 세션에서 바로 꺼내 써도 될 것 같아요.

yuncorn commented on src/main/webapp/resources/js/members/changePassword.js:
```diff
@@ -89,4 +90,8 @@ $(function () {
     });
+
+    $(".change_password_next").on('click', function () {
+        window.location.href = "../../main/main.trip";
```
절대 경로로 수정 부탁드릴께요

Ryumina commented on src/main/webapp/resources/js/common/checkPopUp.js:
```diff
@@ -0,0 +1,23 @@
+}
+
+function popUpCancel() {
+    const popupWrap = $(".check_popup_wrap");
```
단지 팝업을 닫는 기능만 한다면
변수를 선언하지 않고 바로 hide 해주는 것이 좋을 것 같습니다.
popUpOk 부분도 동일합니다!

Ryumina commented on src/main/webapp/resources/js/members/mypage.js:
```diff
@@ -0,0 +1,30 @@
+    });
+
+    $(".mypage_change_password_wrap").on('click', function () {
+        window.location.href = "../mypage/changePassword.trip";
```
단순히 페이지만 이동시키는 로직이 중복되는 것 같은데
url을 인자로 받는 페이지 이동을 위한 함수를 따로 추가하여 해당 함수만 호출할 수 있도록
리팩토링 해주시면 좋을 것 같습니다.

cjyeon committed:
- popUpOpen method chaining 적용
- common.js 생성 > 페이지 이동 메서드 추가 및 적용
- mypage 회원정보 노출하도록 수정
- 상대경로 절대경로로 수정
- ajax error case -> console.error()로 수정

cjyeon committed:
- 이메일 인증 url 수정
- 이메일 변경 메일 인증 시 > 세션 email 업데이트 로직 추가

freesens commented on src/main/webapp/resources/js/members/changeEmail.js:
```diff
@@ -60,7 +60,7 @@ $(function () {
             },
             error: function (error) {
-                console.log(error);
+                console.error(error);
```
* console.error를 사용하면 콘솔창에 에러로 표시되는데, 개발자의 디버깅 목적이라면 상관없지만, 일반 고객들이 보는 화면에 무조건 console.error를 기록하거나, console.log로 불필요하게 상세한 내역을 기록하지 않도록 고민 해주세요.

freesens commented on src/main/java/com/tripmate/controller/MemberController.java:
```diff
@@ -22,7 +22,6 @@
 import retrofit2.Call;
```
* P/R 리뷰 대상 파일이 31개로 너무 많네요.
* 작은 업무단위로 P/R 생성해서 리뷰 완료 후 다음 작업을 시작 해주세요.
* 한번에 리뷰 대상 파일이 지금과 같이 너무 많으면 로직도 복잡하고, 봐야되는 내용도 많아서 흐름을 따라가면서 리뷰하기 곤란해 보입니다.
* 지금과 같이 계속 대량의 파일이 추가되는 형태라면 리뷰 해주는 사람들도 따라가기 어려어서 리뷰를 못해주니, 오히려 작업 일정이 더 지연될 것으로 보입니다.
* 리뷰가 지연된다고 다른 업무까지 추가로 작업해서 계속 리뷰 대상 파일 수를 늘리지 말고, 기존 건 리뷰 내용만 대응해서 수정하고, 빠른 리뷰를 재요청 해서 머지가 완료되도록 해주세요.

cjyeon committed:
- console 출력 코드 삭제
